moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -55,7 +56,18 @@ const validationSchema = yup.object().shape({
   phase: yup.object().nullable().optional(),
   subphase: yup.object().nullable().optional(),
   tags: yup.array().optional(),
-  completionDate: yup.date().nullable().optional(),
+  completionDate: yup
+    .date()
+    .nullable()
+    .optional()
+    .when("phase", {
+      is: (phase) => isPhaseOptionSimpleComplete(phase),
+      then: yup
+        .date()
+        .required(
+          "Required if a phase with phase name simple of Complete is selected"
+        ),
+    }),
```

Comment:
🔥 yup is so great and so are you for doing cool things with it 👍

Reply:
i've been wanting to see what the fuss is about. Yup does typescript stuff too, fwiw, but it also seems like it's just barely maintained by one person. Zod looks great.